src/foam/nanos/crunch/JavaCrunchService.java

```diff
@@ -0,0 +1,82 @@
+package foam.nanos.crunch;
+
+import foam.core.X;
+import foam.dao.ArraySink;
+import foam.dao.DAO;
+import foam.nanos.logger.Logger;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import static foam.mlang.MLang.*;
+
+public class JavaCrunchService implements CrunchService {
+  public List getGrantPath(X x, String rootId) {
+    Logger logger = (Logger) x.get("logger");
+    logger.info("Handling CrunchService grantPath request: " + rootId);
+
+    DAO prerequisiteDAO = (DAO) x.get("prerequisiteCapabilityJunctionDAO");
+    DAO capabilityDAO = (DAO) x.get("capabilityDAO");
+    // TODO:
+    // DAO junctionDAO
```

Comment:
can you be more specific on this and the below todo - please - has me scratching my head

Reply:
oops, todo comment was for me and no longer applies